specs/ledger/hs/src/Ledger/Core.hs

```diff
@@ -112,23 +113,67 @@ minusSlot (Slot m) (SlotCount n)
 -- Domain restriction and exclusion
 ---------------------------------------------------------------------------------
 
--- |Domain restriction
---
-(◁), (◃) :: Ord a => Set a -> Map a b -> Map a b
-s ◁ r = Map.filterWithKey (\k _ -> k `Set.member` s) r
-s ◃ r = s ◁ r
+newtype PairSet a b = PairSet {unPairSet :: Set (a,b)}
+  deriving (Eq, Show)
 
--- |Domain exclusion
---
-(⋪) :: Ord a => Set a -> Map a b -> Map a b
-s ⋪ r = Map.filterWithKey (\k _ -> k `Set.notMember` s) r
+psSize :: PairSet a b -> Int
+psSize = Set.size . unPairSet
 
+class Maplike m where
+  singleton :: a -> b -> m a b
 
--- | Range restriction
---
-(▹) :: Ord b => Map a b -> Set b -> Map a b
-r ▹ s = Map.filter (flip Set.member s) r
+  -- | Domain
+  dom :: Ord a => m a b -> Set a
+  -- | Range
+  range :: Ord b => m a b -> Set b
```

Comment:
why not `ran` ? (Or the other way round, why not `domain`?)

Reply:
`dom` / `ran` is used in Event-B for this btw.

Reply:
No grand reason for this other than I'm not used to `ran` as a shorthand for `range`, whereas `dom` I am used to :-)